tests/test_custom_staleness.py

```diff
@@ -360,3 +360,63 @@ def test_async_update_host_update_custom_staleness(
 
             # Call event_producer
             assert event_producer.write_event.call_count == num_hosts
+
+
+@pytest.mark.parametrize("num_hosts", [1, 2, 3])
+def test_async_update_host_update_custom_staleness_no_modified_on_change(
+    db_create_staleness_culling,
+    db_get_hosts,
+    db_create_multiple_hosts,
+    api_get,
+    api_patch,
+    flask_app,
+    event_producer,
+    mocker,
+    num_hosts,
+):
+    db_create_staleness_culling(**CUSTOM_STALENESS_HOST_BECAME_STALE)
+    with (
+        patch("app.models.get_flag_value", return_value=True),
+        patch("app.serialization.get_flag_value", return_value=True),
+        patch("app.staleness_serialization.get_flag_value", return_value=True),
+        patch("api.host_query_db.get_flag_value", return_value=True),
+        patch("api.staleness.get_flag_value", return_value=True),
+        patch("app.models.datetime") as mock_datetime,
+    ):
+        with flask_app.app.app_context():
+            mocker.patch.object(event_producer, "write_event")
+            _now = now()
+            mock_datetime.now.return_value = _now
+            created_hosts = db_create_multiple_hosts(how_many=num_hosts)
+            host_url = build_hosts_url()
+            response_status, response_data = api_get(host_url)
+            assert response_status == 200
+            assert len(response_data["results"]) == len(created_hosts)
+
+            host_ids = [host["id"] for host in response_data["results"]]
+            hosts_before_update = db_get_hosts(host_ids).all()
```

Comment:
IDK if it's necessary to validate the `GET /hosts` API response in this test. You can probably just simplify it to something like this
```suggestion
            hosts_before_update = db_create_multiple_hosts(how_many=num_hosts)
```

Reply:
Thanks, I have simplified the test